packages/salesforcedx-vscode-core/src/commands/util/postconditionCheckers.ts

```diff
@@ -173,3 +177,96 @@ export class OverwriteComponentPrompt
     return choices;
   }
 }
+
+export interface ConflictDetectionMessages {
+  warningMessageKey: string;
+  commandHint: (input: string) => string;
+}
+
+export class ConflictDetectionChecker implements PostconditionChecker<string> {
+  private messages: ConflictDetectionMessages;
+
+  public constructor(messages: ConflictDetectionMessages) {
+    this.messages = messages;
+  }
+
+  public async check(
+    inputs: ContinueResponse<string> | CancelResponse
+  ): Promise<ContinueResponse<string> | CancelResponse> {
+    if (!sfdxCoreSettings.getConflictDetectionEnabled()) {
+      return inputs;
+    }
+
+    if (inputs.type === 'CONTINUE') {
+      const usernameOrAlias = await this.getDefaultUsernameOrAlias();
+      if (!usernameOrAlias) {
+        return {
+          type: 'CANCEL',
+          msg: nls.localize('conflict_detect_no_default_username')
+        };
+      }
+
+      const defaultPackageDir = await this.getDefaultPackageDir();
+      if (!defaultPackageDir) {
+        return {
+          type: 'CANCEL',
+          msg: nls.localize('conflict_detect_no_default_package_dir')
+        };
+      }
+
+      const config: ConflictDetectionConfig = {
+        usernameOrAlias,
+        packageDir: defaultPackageDir,
+        manifest: inputs.data
+      };
+      const results = await conflictDetector.checkForConflicts(config);
+
+      if (results.different.size > 0) {
+        channelService.appendLine(
+          nls.localize('conflict_detect_conflict_header')
+        );
+        results.different.forEach(file => {
+          channelService.appendLine(file);
+        });
+        channelService.showChannelOutput();
+
+        const choice = await notificationService.showWarningModal(
+          nls.localize(this.messages.warningMessageKey),
+          nls.localize('conflict_detect_override')
+        );
+
+        if (choice !== nls.localize('conflict_detect_override')) {
+          channelService.appendLine(
+            nls.localize(
+              'conflict_detect_command_hint',
+              this.messages.commandHint(inputs.data)
+            )
+          );
+          channelService.showChannelOutput();
+          return { type: 'CANCEL' };
+        }
+      }
+      return inputs;
+    }
+    return { type: 'CANCEL' };
+  }
+
+  public async getDefaultUsernameOrAlias(): Promise<string | undefined> {
+    return await OrgAuthInfo.getDefaultUsernameOrAlias(true);
+  }
+
+  public async getDefaultPackageDir(): Promise<string | undefined> {
```

Comment:
Seems like a useful helper that could live in the `SfdxPackageDirectories` util, in case future code needs the default package directory.

Reply:
Done